src/app/src/main/kotlin/org/imperial/mrc/hint/controllers/ADRController.kt

```diff
@@ -151,4 +158,48 @@ class ADRController(private val encryption: Encryption,
     {
         return saveAndValidate(url, FileType.ANC)
     }
+
+    @PostMapping("/datasets/{id}/resource/{resourceType}/{modelCalibrateId}")
+    @Suppress("ReturnCount")
+    fun pushFileToADR(@PathVariable id: String,
+                      @PathVariable resourceType: String,
+                      @PathVariable modelCalibrateId: String,
+                      @RequestParam resourceFileName: String,
+                      @RequestParam resourceId: String?): ResponseEntity<String>
+    {
+        // 1. Download relevant artefact from hintr
+        val artefact = when (resourceType)
+        {
+            appProperties.adrOutputZipSchema -> Pair(apiClient.downloadSpectrum(modelCalibrateId),
+                    "Naomi model outputs")
+            appProperties.adrOutputSummarySchema -> Pair(apiClient.downloadSummary(modelCalibrateId),
+                    "Naomi summary report")
+            else -> return ErrorDetail(HttpStatus.BAD_REQUEST, "Invalid resourceType").toResponseEntity()
+        }
+
+        // 2. Return error if artefact can't be retrieved
+        if (!artefact.first.statusCode.is2xxSuccessful)
+        {
+            val baos = ByteArrayOutputStream()
+            artefact.first.body?.writeTo(baos)
+            return ErrorDetail(artefact.first.statusCode, baos.toString()).toResponseEntity()
+        }
+
+        // 3. Stream artefact to file
+        val file = File(Files.createTempDirectory("adr").toFile(), resourceFileName).apply { deleteOnExit() }
```

Comment:
This will delete the file at the end of the request but leave the dir behind? That's what I'm seeing locally anyway. Shouldn't cause a problem.

Reply:
I've switched this to delete the folder, after the file has been uploaded